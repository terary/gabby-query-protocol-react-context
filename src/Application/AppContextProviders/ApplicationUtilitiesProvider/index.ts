import ApplicationUtilitiesProvider, {
  ApplicationUtilitiesContext,
  useTestUtilities,
} from "./ApplicationUtilitiesProvider";
import type { ApplicationUtilitiesType } from "./type";

export default ApplicationUtilitiesProvider;
export { ApplicationUtilitiesContext, useTestUtilities };
export type { ApplicationUtilitiesType };
