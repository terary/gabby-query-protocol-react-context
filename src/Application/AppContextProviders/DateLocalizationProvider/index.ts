import DateLocalizationProvider from "./Provider";

export default DateLocalizationProvider;
