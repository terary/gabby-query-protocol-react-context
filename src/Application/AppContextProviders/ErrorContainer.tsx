import { ReactNode } from "react";
import { ErrorBoundary } from "react-error-boundary";

function ErrorFallback({ error }: { error: Error }) {
  return (
    <div role="alert">
      <p>Something went wrong:</p>
      <pre>{error.message}</pre>
    </div>
  );
}

export const ErrorContainer = ({ children }: { children: ReactNode }) => {
  return <ErrorBoundary FallbackComponent={ErrorFallback}>{children}</ErrorBoundary>;
};
