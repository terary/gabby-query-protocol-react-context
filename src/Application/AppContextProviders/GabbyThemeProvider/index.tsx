import { GabbyThemeProvider } from "./Provider";
export default GabbyThemeProvider;
