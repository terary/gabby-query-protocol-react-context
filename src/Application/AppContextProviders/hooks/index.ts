import { useApplicationUtilities } from "./useApplicationUtilities";

export { useApplicationUtilities };
