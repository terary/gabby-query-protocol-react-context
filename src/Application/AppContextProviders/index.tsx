import React from "react";
import { TSupportedLocales } from "..";
import ApplicationUtilitiesProvider, {
  useTestUtilities,
} from "./ApplicationUtilitiesProvider";
import { ErrorContainer } from "./ErrorContainer";
import DateLocalizationProvider from "./DateLocalizationProvider";
import GabbyThemeProvider from "./GabbyThemeProvider";

interface IAppProviderParameters {
  children?: React.ReactNode;
  initialLocaleCode?: TSupportedLocales;
}

export const AppContextProviders = ({
  initialLocaleCode,
  children,
}: IAppProviderParameters) => {
  return (
    <ErrorContainer>
      {" "}
      {/* Error container may have issue with hooks, requires testing */}
      <ApplicationUtilitiesProvider initialLocaleCode={initialLocaleCode}>
        <GabbyThemeProvider>
          <DateLocalizationProvider>{children}</DateLocalizationProvider>
        </GabbyThemeProvider>
      </ApplicationUtilitiesProvider>
    </ErrorContainer>
  );
};

export { useTestUtilities };
