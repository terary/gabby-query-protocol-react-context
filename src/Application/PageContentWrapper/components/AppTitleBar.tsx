/* eslint-disable import/prefer-default-export */
import Toolbar from "@mui/material/Toolbar";
import Typography from "@mui/material/Typography";
import Badge from "@mui/material/Badge";
import MenuIcon from "@mui/icons-material/Menu";
import NotificationsIcon from "@mui/icons-material/Notifications";
import IconButton from "@mui/material/IconButton";
import { AppTitleBarStyled } from "./AppTitleBarStyled";
import { useApplicationUtilities } from "../../AppContextProviders/hooks";
import { LanguageSwitcher } from "./LanguageSwitcher";
interface IAppBarProps {
  isOpen: boolean;
  titleText: string;
  toggleDrawer: () => void;
}
export const AppTitleBar = ({
  isOpen,
  titleText,
  toggleDrawer,
}: IAppBarProps): JSX.Element => {
  const { t, getLocale, getCurrentLocale } = useApplicationUtilities();
  const currentLocale = getCurrentLocale();
  const RtlToolBar = () => {
    return (
      <Toolbar
        sx={{
          pr: "24px", // keep right padding when drawer closed
        }}
      >
        <IconButton color="inherit">
          <Badge badgeContent={4} color="secondary">
            <NotificationsIcon />
          </Badge>
        </IconButton>
        <Typography component="h1" variant="h6" color="inherit" noWrap sx={{ flexGrow: 1 }}>
          {titleText} <br />
          {t("app-title-bar-text")}
          <br />
        </Typography>
        <LanguageSwitcher />
        <IconButton
          edge="start"
          color="inherit"
          aria-label="open drawer"
          onClick={toggleDrawer}
          sx={{
            marginLeft: "36px",
            ...(isOpen && { display: "none" }),
          }}
        >
          <MenuIcon />
        </IconButton>
      </Toolbar>
    );
  };

  const LtrToolBar = () => {
    return (
      <Toolbar
        sx={{
          pr: "24px", // keep right padding when drawer closed
        }}
      >
        <IconButton
          edge="start"
          color="inherit"
          aria-label="open drawer"
          onClick={toggleDrawer}
          sx={{
            marginRight: "36px",
            ...(isOpen && { display: "none" }),
          }}
        >
          <MenuIcon />
        </IconButton>
        <Typography component="h1" variant="h6" color="inherit" noWrap sx={{ flexGrow: 1 }}>
          {titleText} <br />
          {t("app-title-bar-text")}
          <br />
        </Typography>
        <LanguageSwitcher />
        <IconButton color="inherit">
          <Badge badgeContent={4} color="secondary">
            <NotificationsIcon />
          </Badge>
        </IconButton>
      </Toolbar>
    );
  };

  return (
    <AppTitleBarStyled open={isOpen} dir="ltr">
      {currentLocale.languageDir === "ltr" && <LtrToolBar />}
      {currentLocale.languageDir === "rtl" && <RtlToolBar />}
    </AppTitleBarStyled>
  );
};
