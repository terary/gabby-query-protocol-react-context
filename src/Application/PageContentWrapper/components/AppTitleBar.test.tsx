import React, { useState } from "react";
import renderer, { act } from "react-test-renderer";
import { AppTitleBar } from "./AppTitleBar";
import { toMatchDiffSnapshot } from "snapshot-diff";
import { AppContextProviders } from "../../AppContextProviders";
expect.extend({ toMatchDiffSnapshot });

interface IAppBarProps {
  isOpen: boolean;
  titleText: string;
  toggleDrawer: () => void;
}
const AppBarContextWrapper = ({ isOpen, titleText, toggleDrawer }: IAppBarProps) => {
  return (
    <AppContextProviders>
      <AppTitleBar titleText={titleText} isOpen={isOpen} toggleDrawer={() => {}} />
    </AppContextProviders>
  );
};

describe("AppTitleBar", () => {
  describe("Snapshots open/close", () => {
    it("Should render title bar adjusted for left horizontal menu, opened", () => {
      const titleText = "The Awesome Test App";

      const tree = renderer
        .create(
          <AppBarContextWrapper titleText={titleText} isOpen={true} toggleDrawer={() => {}} />
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it("Should render title bar adjusted for left horizontal menu, closed", () => {
      const titleText = "The Awesome Test App";

      const tree = renderer
        .create(
          <AppBarContextWrapper titleText={titleText} isOpen={false} toggleDrawer={() => {}} />
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe("isOpen state managed by parent", () => {
    it("Should change open/closed state by setting 'isOpen'", () => {
      const AppBarStateWrapper = () => {
        const titleText = "The Awesome Test App";
        const [isOpen, setIsOpen] = useState(false);
        const toggleDrawer = () => {
          setIsOpen(!isOpen);
        };
        return (
          <AppBarContextWrapper
            isOpen={isOpen}
            titleText={titleText}
            toggleDrawer={toggleDrawer}
          />
        );
      };

      const component = renderer.create(<AppBarStateWrapper />);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();

      act(() => {
        component.root.findByProps({ "aria-label": "open drawer" }).props.onClick();
        const treeUpdated = component.toJSON();
        expect(tree).toMatchDiffSnapshot(treeUpdated);
      });
    });
  });
});
