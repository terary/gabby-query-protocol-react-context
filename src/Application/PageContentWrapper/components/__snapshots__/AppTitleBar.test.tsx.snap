// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppTitleBar Snapshots open/close Should render title bar adjusted for left horizontal menu, closed 1`] = `
Array [
  " ",
  <div
    className="Toastify"
  />,
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-19z1ozs-MuiPaper-root-MuiAppBar-root"
    dir="ltr"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-14v3fac-MuiToolbar-root"
    >
      <button
        aria-label="open drawer"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeMedium css-dhvns5-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <h1
        className="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-uh02xs-MuiTypography-root"
      >
        The Awesome Test App
         
        <br />
        Algún otro texto de texto
        <br />
      </h1>
      <div>
        <div
          className="MuiFormControl-root css-1869usk-MuiFormControl-root"
        >
          <div
            className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiSelect-root css-163q0ur-MuiInputBase-root-MuiInput-root"
            onClick={[Function]}
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="demo-simple-select-standard-label demo-simple-select-standard"
              className="MuiSelect-select MuiSelect-standard MuiInput-input MuiInputBase-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
              id="demo-simple-select-standard"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              role="button"
              tabIndex={0}
            >
              México
            </div>
            <input
              aria-hidden={true}
              className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              disabled={false}
              onAnimationStart={[Function]}
              onChange={[Function]}
              required={false}
              tabIndex={-1}
              value="es_mx"
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </div>
      <button
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiBadge-root css-1c32n2y-MuiBadge-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="NotificationsIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.89 2 2 2zm6-6v-5c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2z"
            />
          </svg>
          <span
            className="MuiBadge-colorSecondary MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginTopRightRectangular css-jcn4dz-MuiBadge-badge"
          >
            4
          </span>
        </span>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </header>,
]
`;

exports[`AppTitleBar Snapshots open/close Should render title bar adjusted for left horizontal menu, opened 1`] = `
Array [
  " ",
  <div
    className="Toastify"
  />,
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-1191obr-MuiPaper-root-MuiAppBar-root"
    dir="ltr"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-14v3fac-MuiToolbar-root"
    >
      <button
        aria-label="open drawer"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeMedium css-1johsky-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </button>
      <h1
        className="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-uh02xs-MuiTypography-root"
      >
        The Awesome Test App
         
        <br />
        Algún otro texto de texto
        <br />
      </h1>
      <div>
        <div
          className="MuiFormControl-root css-1869usk-MuiFormControl-root"
        >
          <div
            className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiSelect-root css-163q0ur-MuiInputBase-root-MuiInput-root"
            onClick={[Function]}
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="demo-simple-select-standard-label demo-simple-select-standard"
              className="MuiSelect-select MuiSelect-standard MuiInput-input MuiInputBase-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
              id="demo-simple-select-standard"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              role="button"
              tabIndex={0}
            >
              México
            </div>
            <input
              aria-hidden={true}
              className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              disabled={false}
              onAnimationStart={[Function]}
              onChange={[Function]}
              required={false}
              tabIndex={-1}
              value="es_mx"
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </div>
      <button
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiBadge-root css-1c32n2y-MuiBadge-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="NotificationsIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.89 2 2 2zm6-6v-5c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2z"
            />
          </svg>
          <span
            className="MuiBadge-colorSecondary MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginTopRightRectangular css-jcn4dz-MuiBadge-badge"
          >
            4
          </span>
        </span>
      </button>
    </div>
  </header>,
]
`;

exports[`AppTitleBar isOpen state managed by parent Should change open/closed state by setting 'isOpen' 1`] = `
Array [
  " ",
  <div
    className="Toastify"
  />,
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-19z1ozs-MuiPaper-root-MuiAppBar-root"
    dir="ltr"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-14v3fac-MuiToolbar-root"
    >
      <button
        aria-label="open drawer"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeMedium css-dhvns5-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </button>
      <h1
        className="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-uh02xs-MuiTypography-root"
      >
        The Awesome Test App
         
        <br />
        Algún otro texto de texto
        <br />
      </h1>
      <div>
        <div
          className="MuiFormControl-root css-1869usk-MuiFormControl-root"
        >
          <div
            className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiSelect-root css-163q0ur-MuiInputBase-root-MuiInput-root"
            onClick={[Function]}
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="demo-simple-select-standard-label demo-simple-select-standard"
              className="MuiSelect-select MuiSelect-standard MuiInput-input MuiInputBase-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
              id="demo-simple-select-standard"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              role="button"
              tabIndex={0}
            >
              México
            </div>
            <input
              aria-hidden={true}
              className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              disabled={false}
              onAnimationStart={[Function]}
              onChange={[Function]}
              required={false}
              tabIndex={-1}
              value="es_mx"
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </div>
      <button
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiBadge-root css-1c32n2y-MuiBadge-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="NotificationsIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.89 2 2 2zm6-6v-5c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2z"
            />
          </svg>
          <span
            className="MuiBadge-colorSecondary MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginTopRightRectangular css-jcn4dz-MuiBadge-badge"
          >
            4
          </span>
        </span>
      </button>
    </div>
  </header>,
]
`;

exports[`AppTitleBar isOpen state managed by parent Should change open/closed state by setting 'isOpen' 2`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;
